.buildkite/pipeline.yml

```diff
@@ -0,0 +1,18 @@
+steps:
+  - label: "Validate Clang Format"
+    command: "/code/NUbots/.travis/validate_format.sh"
+    plugins:
+      - docker#v3.3.0:
+          image: "nubots/buildkite:0.2"
+          workdir: "/code/NUbots"
+          propagate-uid-gid: true
+  - label: "Build for NUC"
+    command: "/code/NUbots/.travis/build_code.sh"
+    plugins:
+      - docker#v3.3.0:
+          image: "nubots/buildkite:0.2"
+          workdir: "/code/NUbots"
+          propagate-uid-gid: true
+          environment:
+            - "PLATFORM=nuc7i7bnh"
+            - "CCACHE_DIR=/code"
```

Comment:
Does this cache dir persist through different builds? Otherwise, ccache is useless

Reply:
`ccache` is still useful for local builds right? E.g. in the case where your build fails halfway through, you fix an error, run the build again and it doesn't have to start from scratch.

If it isn't ccache that's providing this then yeah, we should probably remove it.

Reply:
The way ccache works is whenever you compile a file, it takes a hash of the content that was compiled. Then if you ever compile the same file again it just gives you the compiled object. But to do this it needs a persistent storage location so that it can store this cache. For what we're doing we should probably remove it.

Reply:
Removed `ccache`.